lib/hiera/backend/vault_backend.rb

```diff
@@ -61,13 +82,21 @@ def lookup_generic(key, scope)
           return nil if secret.nil?
 
           Hiera.debug("[hiera-vault] Read secret: #{key}")
-          if @config[:default_field]
+          if @config[:default_field] and secret.data.has_key?(@config[:default_field].to_sym) and secret.data.length == 1
```

Comment:
Point taken, the example I'm imaging is setup like `:default_field = "hiera_only_value"`.